### PR TITLE
RFC1: Add SourceLocation Specification

### PR DESCRIPTION
**Pull request by ckchen-google**
https://github.com/opencomputeproject/ocp-diag-core/discussions/634
 Signed-off-by: ChunKai Chen <chunkaichen@google.com>

### CODE REVIEW

File: json_spec/output/diagnosis.json [hunk 1]
```diff
@@ -23,7 +23,11 @@
         "subcomponent": {
             "description": "TODO: link to md",
             "$ref": "/opencomputeproject/ocp-diag-core/subcomponent"
-        }
+        },
+	"sourceLocation": {
+            "description": "TODO: link to md doc",
```
Comment (mimir-d) [inline json_spec/output/diagnosis.json:28]: you can actually fill these in now; the TODO was before the spec README.md was available
it should be a "short string to describe the item + link to md doc with #anchor"

File: json_spec/output/source_location.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://github.com/opencomputeproject/ocp-diag-core/sourceLocation",
+    "title": "Source Location",
+    "description": "TODO: link to md doc",
```
Comment (mimir-d) [inline json_spec/output/source_location.json:5]: comment above about filling these in applies everywhere

File: json_spec/README.md [hunk 3]
```diff
@@ -2033,6 +2063,57 @@ Provides information about software that was discovered or exercised during the
...
+    <tr>
+        <td><strong>Attribute</strong></td>
+        <td><strong>Type</strong></td>
+        <td><strong>Required</strong></td>
+        <td><strong>Description</strong></td>
+    </tr>
+    <tr>
+        <td><em>file</em></td>
+        <td>string</td>
+        <td>No</td>
```
Comment (mimir-d) [inline json_spec/README.md:2084]: https://github.com/opencomputeproject/ocp-diag-core/discussions/634 says these are required

File: json_spec/README.md [hunk 4]
```diff
@@ -2033,6 +2063,57 @@ Provides information about software that was discovered or exercised during the
 ```
 
 
+#### SourceLocation
+
+##### Description
+
+Provides information about source location for the purpose of debugging or tracing program execution.
```
Comment (mimir-d) [inline json_spec/README.md:2070]: i still think this isnt explicit enough to say that the reference for these lines/files is the diagnostic package itself

Comment (mimir-d) [reply json_spec/README.md:2070]: same with the individual fields

**Review by mimir-d — CHANGES_REQUESTED** [review]

File: json_spec/output/diagnosis.json (hunk 1 above)
Comment (ckchen-google) [reply json_spec/output/diagnosis.json]: updated

File: json_spec/output/source_location.json (hunk 2 above)
Comment (ckchen-google) [reply json_spec/output/source_location.json]: updated

File: json_spec/README.md (hunk 3 above)
Comment (ckchen-google) [reply json_spec/README.md]: fixed

File: json_spec/README.md (hunk 4 above)
Comment (ckchen-google) [reply json_spec/README.md]: updated

**Review by mimir-d — CHANGES_REQUESTED** [review]

**Comment by mimir-d** [comment]
@arunkoshy waiting for an approval from you; i think we need to add you to the gh team here
@dframe-google are we blocked on the DCO or can we use a blank signature for this one (and we provide the DCO text later)?

**Comment by arunkoshy** [comment]
@mimir-d LGTM based on our discussion.

**Review by mimir-d — APPROVED** [review]

**Comment by mimir-d** [comment]
@ckchen-google thanks for this PR and the fixes. seems the DCO bot isnt liking your signature. we can manually bypass this, but ideally the commits would be fixed.
here's the list of errors:
```
Commit sha: [6c1e522], Author: Chun-Kai Chen, Committer: Chun-Kai Chen; Expected "Chun-Kai Chen [chunkaichen@google.com](mailto:chunkaichen@google.com)", but got "ChunKai Chen [chunkaichen@google.com](mailto:chunkaichen@google.com)".
Commit sha: [6d065cc], Author: ckchen-google, Committer: ckchen-google; Expected "ckchen-google [chunkaichen@google.com](mailto:chunkaichen@google.com)", but got "ChunKai Chen [chunkaichen@google.com](mailto:chunkaichen@google.com)".
Commit sha: [120df12], Author: ckchen-google, Committer: ckchen-google; The sign-off is missing
```

**Comment by arunkoshy** [comment]
@ckchen-google @mimir-d  yep, DCO is " literal " - my recommendation is to setup something automatic in your git workflow to ensure the signature is fixed to what passes going forward. I'd to do this for OCP-OSFCI.

**Comment by ckchen-google** [comment]
Fixed DCO issues.

**Pull request merged**